mllib/src/main/scala/org/apache/spark/ml/feature/MinHashLSH.scala

```diff
@@ -70,19 +70,51 @@ class MinHashLSHModel private[ml](
 
   @Since("2.1.0")
   override protected[ml] def keyDistance(x: Vector, y: Vector): Double = {
-    val xSet = x.nonZeroIterator.map(_._1).toSet
-    val ySet = y.nonZeroIterator.map(_._1).toSet
-    val intersectionSize = xSet.intersect(ySet).size.toDouble
-    val unionSize = xSet.size + ySet.size - intersectionSize
+    val xIter = x.nonZeroIterator.map(_._1)
```

Comment:
Just so I'm clear, this iterator is definitely in order and has unique elements already?

Reply:
Yes, the iterator here is in order and has unique indices